### PR TITLE
Completed challenge 6

### PR DESCRIPTION
**Pull request by 08Reild**
I have completed challenge number 6. I have added a get.all to my app.js for all invalid paths and I have deleted the relevant code from my getTopics function/fetchTopics function so this only handles the returning of the data and my get.all handled the error. Thank you

### CODE REVIEW

**Review by yalicee — CHANGES_REQUESTED** [review]
Please add the following tests:

Status 200, valid ID, but has no comments responds with an empty array of comments
Status 400, invalid ID, e.g. string of "not-an-id"

**Comment by 08Reild** [comment]
hey @yalicee, does my previous test above not cover the status 400, invalid ID?

my test on line 32 in errors.test.js is:

test("Requesting articles with a non numeric value instead of a article_id number returns 400 bad request", () => {
        return request(app)
            .get('/api/articles/invalidrequest')
            .expect(400)
            .then(({ body }) => {
                expect(body.msg).toBe("Bad Request")
            })
    })


Thanks :)

**Comment by yalicee** [comment]
> hey @yalicee, does my previous test above not cover the status 400, invalid ID?
> 
> 
> 
> my test on line 32 in errors.test.js is:
> 
> 
> 
> test("Requesting articles with a non numeric value instead of a article_id number returns 400 bad request", () => {
> 
>         return request(app)
> 
>             .get('/api/articles/invalidrequest')
> 
>             .expect(400)
> 
>             .then(({ body }) => {
> 
>                 expect(body.msg).toBe("Bad Request")
> 
>             })
> 
>     })
> 
> 
> 
> 
> 
> Thanks :) 

No because that's not for comments  on articles

**Comment by 08Reild** [comment]
Hey @yalicee I have amended the file with the edits :) do I need to submit a new PR? Thank you!

**Comment by 08Reild** [comment]
Sorry @yalicee , I just remembered I do so ignore that last message. Ive submitted the pr:)

File: Controllers/controllers.js [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
-    .then((comments) => {
-        res.status(200).send({comments: comments})
-    })
-    .catch((err) => {
-        //console.log(err, "I'm the error in the getArticlesComment function in the controller")
-        next(err)
+    fetchArticlesById(article_id).then((result) => {
+        return fetchArticlesComments(article_id)
+            .then((comments) => {
+                res.status(200).send({ comments: comments })
```
Comment (itskatherine) [inline Controllers/controllers.js:44]: FYI if you have an object where you're giving it a key and you're assigning it a value that's a variable of the same name (like you're doing here with `comments`), you can use the shorthand {comments}
So
`{ comments: comments }`
is equivalent to
`{comments}`

(This isn't a mandatory change just worth knowing!) 🧠

**Review by itskatherine — APPROVED** [review]
Good to merge Dan! 🙌

**Pull request merged**